src/mercury-inventory/mercury/inventory/db.py

```diff
@@ -60,18 +63,30 @@ def __init__(self, collection):
         :return:
         """
         data = data or {}
+        upsert = False
 
         if not data:
             log.warning('Update data is empty')
 
-        if not await self.collection.count({'mercury_id': mercury_id}, projection={'_id': 1}):
+        exists = await self.collection.count({'mercury_id': mercury_id}, projection={'_id': 1})
+        if not upsert and not exists:
             raise MercuryGeneralException('Attempting to update non-existing record')
 
         if 'mercury_id' in data:
-            raise MercuryCritical('MercuryID cannot be updated once a record already exists')
+            if upsert:
+                del data['mercury_id']
+            else:
+                raise MercuryCritical('MercuryID cannot be updated once a record already exists')
+
+        hooks = await get_hooks_from_data(data)
```

Comment:
The hooks that are implemented operate on data structures that are already loaded into memory, and do not require any IO operation. The act of adding them to the event loop is probably more expensive than running them synchronously. Unless I am missing something, we should probably convert get_hooks (and subsequently called coroutines) to standard functions